LibraryManagementSystem/Library.cs

```diff
@@ -35,52 +31,51 @@ public void ViewBooks()
         }
 
         //helper method to find books
-        public Book FindBookById(int id)
+        public Book? FindBookById(int id)
         {
-            foreach (var book in books)
-            {
-                if (book.ID == id)
-                {
-                    return book;
-                }
-            }
-            return null;
+            return books.Find(b => b.ID == id);
         }
 
 
-        public void BorrowBook(int id)
+        public bool BorrowBook(int id)
         {
-            Book book = FindBookById(id);
+            Book? book = FindBookById(id);
             if (book == null)
             {
-                Console.WriteLine("\nBook not found");
+                Console.WriteLine("\nBook not found.");
+                return false;
             }
             else if (book.IsBorrowed)
             {
-                Console.WriteLine("\nBook is already borrowed");
+                Console.WriteLine("\nBook is already borrowed.");
+                return false;
             }
             else
             {
                 book.IsBorrowed = true;
-                Console.WriteLine("\nBook borrowed successfully");
+                Console.WriteLine("\nBook borrowed successfully.");
+                return true;
             }
         }
 
```

Comment:
Can we consider using a `(string, bool)` return type so we can avoid logging two messages to the client on failure?